new-backend/app/modules/langgraph_nodes/store_and_send.py

```
@@ -1,12 +1,26 @@
+        try:
+            chunks = chunk_rag_data(state)
+        except KeyError as e:
+            raise Exception(f"Missing required data field for chunking: {e}")
+        except Exception as e:
+            raise Exception(f"Failed to chunk data: {e}")
+        try:
+            vectors = embed_chunks(chunks)
+            if vectors:
+                print("embedding generated successfully!")
+        except Exception as e:
+            raise Exception(f"failed to embed chunks: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)